src/base/static/js/views/layer-view.js

```diff
@@ -24,16 +23,52 @@
       this.layerIsAdminControlled = Util.getAdminStatus(this.model.get("datasetId"));
 
       // On map move, adjust the visibility of the markers for max efficiency
-      this.map.on('move', this.throttledRender, this);
+      this.map.on('moveend', this.throttledRender, this);
+
+      this.placeType = this.options.placeTypes[this.model.get('location_type')];
+      this.styleRuleContext = {};
+      this.styleRules = [];
+      this.zoomRules = [];
+
+      // Create arrays of functions representing parsed versions of style rules
+      // found in the config. This prevents us from having to re-parse each
+      // style rule on every map zoom for every layer view.
+      _.each(this.placeType.rules, function(rule) {
+        var condition = rule.condition.replace(/\{\{/g, "this.").replace(/\}\}/g, ""),
+            fn = new Function("return " + condition + ";");
+
+        fn.props = {};
```

Comment:
I don't think we should be hanging out `props` object off of our Function. Instead, can we define a `var props = {}` and add our placetype attributes to it?

Then we can do something like this:
```
var props = {};
_.each(rule, function(prop, key) {
  if (prop === "condition") {
    props.condition = new Function("return " + rule.condition + ";");
  } else {
    props[key] = rule[key];
  }
}, this);
```